modules/apps/web-experience/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/messaging/ResourcesImporterHotDeployMessageListener.java

```diff
@@ -73,6 +77,16 @@
 
 	@Activate
 	protected void activate(final BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		_messageQueue = new LinkedList<>();
+
+		ServiceTrackerFactory.open(
+			_bundleContext,
+			"(objectClass=com.liferay.portal.osgi.web.portlet.tracker." +
+				"internal.PortletTracker)",
+			new PortletTrackerServiceTrackerCustomizer());
+
 		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
```

Comment:
I'm not even sure why you didn't implement this as a `@Reference` to the PortletTracker? I'm not even sure that it's a good idea to wait for the portletTracker? Are you sure you're not waiting for some concrete implementations of portlets? What if those portlets are not installed?